relay-event-schema/src/protocol/clientsdk.rs

```diff
@@ -12,6 +16,124 @@ pub struct ClientSdkPackage {
     pub version: Annotated<String>,
 }
 
+/// An error returned when parsing setting values fail.
+#[derive(Debug, Clone, Error)]
+pub enum ParseSettingError {
+    #[error("Invalid value for 'infer_ip'.")]
+    InferIp,
+}
+
+/// A collection of settings that are used to control behaviour in relay through flags.
+///
+/// The settings aim to replace magic values in fields which need special treatment,
+/// for example `{{auto}}` in the user.ip_address. The SDK would instead send `auto_infer_ip`
+/// to toggle the behaviour.
+#[derive(Debug, Clone, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+pub struct ClientSdkSettings {
+    infer_ip: Annotated<AutoInferSetting>,
+}
+
+impl ClientSdkSettings {
+    /// Returns the current [`AutoInferSetting`] setting.
+    ///
+    /// Defaults to [`AutoInferSetting::Legacy`].
+    pub fn infer_ip(&self) -> AutoInferSetting {
+        self.infer_ip.value().copied().unwrap_or_default()
```

Comment:
As discussed we can check here for if `infer_ip` contains an error, then default to never otherwise to legacy.